SomerenUI/BaseForm.cs

```diff
@@ -72,9 +72,11 @@ protected void ShowError(string errorMessage, Exception ex)
             MessageBox.Show(errorMessage + ex.Message);
         }
 
-        protected void ShowMessage(string message, string arg = null)
+        protected void ShowMessage(string message, string arg = null, Form form = null)
```

Comment:
It breaks the single responsibility rule and reusability of this method. We could rename it to ShowMessageAndCloseForm but this method is a common case and we might use it further without the closing part. So, I would keep it as it is.